daemon/logger/awslogs/cloudwatchlogs.go

```diff
@@ -482,18 +482,18 @@ func (l *logStream) create() error {
 
 // createLogGroup creates a log group for the instance of the awslogs logging driver
 func (l *logStream) createLogGroup() error {
-	if _, err := l.client.CreateLogGroup(&cloudwatchlogs.CreateLogGroupInput{
+	if _, err := l.client.CreateLogGroup(context.TODO(), &cloudwatchlogs.CreateLogGroupInput{
 		LogGroupName: aws.String(l.logGroupName),
 	}); err != nil {
-		if awsErr, ok := err.(awserr.Error); ok {
+		var apiErr smithy.APIError
+		if errors.As(err, &apiErr) {
 			fields := logrus.Fields{
-				"errorCode":      awsErr.Code(),
-				"message":        awsErr.Message(),
-				"origError":      awsErr.OrigErr(),
+				"errorCode":      apiErr.ErrorCode(),
+				"message":        apiErr.ErrorMessage(),
 				"logGroupName":   l.logGroupName,
 				"logCreateGroup": l.logCreateGroup,
 			}
-			if awsErr.Code() == resourceAlreadyExistsCode {
+			if _, ok := apiErr.(*types.ResourceAlreadyExistsException); ok {
```

Comment:
We should prefer `errors.As` here, with a `var` declaration for `apiErr`.

Reply:
`errors.As` would be subtly incorrect as it could further unwrap the `apiErr`, potentially resulting in a different error being matched than the one being logged. While unlikely in this particular case, directly asserting the type of `apiErro` also better expresses the intent that no further inspection of the error chain is desired.